lib/fs.js

```diff
@@ -2051,7 +2053,29 @@ function writeAll(fd, isUserFd, buffer, offset, length, signal, callback) {
     }
     return;
   }
-  // write(fd, buffer, offset, length, position, callback)
+  if (isCustomIterable(buffer)) {
+    (async () => {
+      for await (const buf of buffer) {
+        fs.write(
+          fd, buf, undefined,
+          isArrayBufferView(buf) ? buf.byteLength : encoding,
+          null, (writeErr, _) => {
+            if (writeErr) {
+              if (isUserFd) {
+                callback(writeErr);
+              } else {
+                fs.close(fd, (err) => {
+                  callback(aggregateTwoErrors(err, writeErr));
+                });
+              }
+            }
+          }
+        );
+      }
+      fs.close(fd, callback);
```

Comment:
The close here is problematic because writes may still be pending from the loop.